### PR TITLE
[Doc][SYCL]  Remove low-level implementation details from the doc

### PR DESCRIPTION
**Pull request by Fznamznon**
Signed-off-by: Mariya Podchishchaeva <mariya.podchishchaeva@intel.com>

### CODE REVIEW

**Comment by Fznamznon** [comment]
Aligned the doc with actual behavior.

File: sycl/doc/SYCLCompilerAndRuntimeDesign.md [hunk 1]
```diff
@@ -154,22 +154,26 @@ targets like SPIR-V).
...
 
-* IsSYCL : True if the user has passed `-fsycl-device-only` to the compilation
-* IsSYCLOffloadDevice: True if calling clang to set up a device compilation
-* IsSYCLHost: True if setting up a call to clang to do a host compilation
+* IsSYCL : True if the user has passed `-fsycl` to the compilation
+* IsSYCLOffloadDevice: True if setting up a call to clang to do a device
+compilation and user has passed `-fsycl` option to the compilation
+* IsSYCLDevice : True if the user has passed `-fsycl-device-only` option to the
+driver. Also true when IsSYCLOffloadDevice is true
+
```
Comment (mdtoguchi) [inline sycl/doc/SYCLCompilerAndRuntimeDesign.md]: Changes LGTM, but is it common to add implementation details in a doc like this?

Comment (Fznamznon) [reply sycl/doc/SYCLCompilerAndRuntimeDesign.md]: I don't know, just fixed confusing information.

Comment (bader) [reply sycl/doc/SYCLCompilerAndRuntimeDesign.md]: I think @mdtoguchi has a point. 
No idea why variables names from Driver implementation is added to this document, but it doesn't seem like a good idea.

Comment (Fznamznon) [reply sycl/doc/SYCLCompilerAndRuntimeDesign.md]: Do you suggest removing them?

Comment (bader) [reply sycl/doc/SYCLCompilerAndRuntimeDesign.md]: Yes.
We should keep the high-level design information about different SYCL modes supported by the driver, but remove low-level implementation details like "bool variables with specific names".

Comment (Fznamznon) [reply sycl/doc/SYCLCompilerAndRuntimeDesign.md]: Okay, done.

File: sycl/doc/SYCLCompilerAndRuntimeDesign.md [hunk 2]
```diff
@@ -154,22 +154,16 @@ targets like SPIR-V).
 
 #### Enable SYCL offload
 
-To enable compilation following the SYCL specification, a special option
+To enable compilation following single-source multiple compiler-passes (SMCP)
+technique which described in the SYCL specification, a special option
```
Comment (AGindinson) [inline sycl/doc/SYCLCompilerAndRuntimeDesign.md]:
```suggestion
technique which is described in the SYCL specification, a special option
```

**Pull request merged**